wagtail/core/tests/test_sites.py

```diff
@@ -56,39 +58,207 @@ def test_no_site_name_default_port_n(self):
+        # requests to the default host on an unrecognised port should be directed to the default site
+        request = HttpRequest()
+        request.path = '/'
+        request.META['HTTP_HOST'] = self.default_site.hostname
+        request.META['SERVER_PORT'] = self.unrecognised_port
+        with self.assertNumQueries(self.find_for_request_queries_expected):
+            self.assertEqual(Site.find_for_request(request), self.default_site)
+
+    def test_unrecognised_port_and_unrecognised_host_routes_to_default_site(self):
+        # requests with an unrecognised Host: header _and_ an unrecognised port
+        # hould be directed to the default site
+        request = HttpRequest()
+        request.path = '/'
+        request.META['HTTP_HOST'] = self.unrecognised_hostname
+        request.META['SERVER_PORT'] = self.unrecognised_port
+        with self.assertNumQueries(self.find_for_request_queries_expected):
+            self.assertEqual(Site.find_for_request(request), self.default_site)
+
+    def test_unrecognised_port_on_known_hostname_routes_there_if_no_ambiguity(self):
+        # requests on an unrecognised port should be directed to the site with
+        # matching hostname if there is no ambiguity
+        request = HttpRequest()
+        request.path = '/'
+        request.META['HTTP_HOST'] = self.about_site.hostname
+        request.META['SERVER_PORT'] = self.unrecognised_port
+        with self.assertNumQueries(self.find_for_request_queries_expected):
+            self.assertEqual(Site.find_for_request(request), self.about_site)
+
+    def test_unrecognised_port_on_known_hostname_routes_to_default_site_if_ambiguity(self):
+        # requests on an unrecognised port should be directed to the default
+        # site, even if their hostname (but not port) matches more than one
+        # other entry
+        request = HttpRequest()
+        request.path = '/'
+        request.META['HTTP_HOST'] = self.events_site.hostname
+        request.META['SERVER_PORT'] = self.unrecognised_port
+        with self.assertNumQueries(self.find_for_request_queries_expected):
+            self.assertEqual(Site.find_for_request(request), self.default_site)
+
+    def test_port_in_http_host_header_is_ignored(self):
+        # port in the HTTP_HOST header is ignored
+        request = HttpRequest()
+        request.path = '/'
+        request.META['HTTP_HOST'] = "%s:%s" % (self.events_site.hostname, self.events_site.port)
+        request.META['SERVER_PORT'] = self.alternate_port_events_site.port
+        with self.assertNumQueries(self.find_for_request_queries_expected):
+            self.assertEqual(Site.find_for_request(request), self.alternate_port_events_site)
+
+
+@override_settings(
+    WAGTAIL_SITE_CACHE_ENABLED=True,
+    CACHES={
+        'default': {
+            'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+            'LOCATION': '/var/tmp/django_cache',
```

Comment:
This may be an overabundance of caution, but does defining the cache location like this require write access in order to run the tests? I notice that [the Django tests of `FileBasedCache`](https://github.com/django/django/blob/acc989f037f4185b3f55cb76ab51c5224f30f0d1/tests/cache/tests.py#L1337) use a temporary directory instead of something hardcoded.

Reply:
@chosak The tests seemed to run fine on Travis CI, but I'm honestly unsure on what the best option here would be. Happy to change to whatever you feel would is suitable.

Reply:
How would you feel about copying that Django logic, to use a temp directory generated at test runtime? I suspect this test would fail on a Windows machine or somewhere without write privileges to `/var/tmp/django_cache`.

Reply:
@chosak If FileBasedCache and LocMemCache are equally unsuitable in terms of running in multiple processes, then I suppose there's no reason why we shouldn't just use LocMemCache as the backend to show the cache working, and avoid complicating the tests any more than needed. Does that sound okay to you?

Reply:
Yes, that sounds like a great idea!